### ISSUE TITLE
Native npm module node-sodium cross compiled to Android 

### ISSUE THREAD

**Issue by gardner** [jxcore/jxcore#880]
What is the best approach for getting an npm module with native code cross compiled for use on Android?

We are trying to get the npm module [node-sodium](https://github.com/paixaop/node-sodium) into the runtime for use within and Android application. The problem is when doing an `npm install` from the host machine and then compiling jxcore with the android-ndk the npm module is compile for the host platform's architecture.

Should we compile the library directly into the jxcore runtime and port the javascript to work from that level? Should we update the npm module to look for an ARCH environment variable and cross compile for Android? Does jxcore for android support shared libraries as such?

I have been having trouble finding a similar example except for leveldown which is why I [started porting](https://github.com/gardner/jxcore/tree/feature/libsodium) libsodium into the jxcore runtime. It would be good to hear from someone with higher level view of the architecture if this is the appropriate approach.

Thank you,
  Gardner

**Comment by lamuertepeluda** [comment]
Have you tried the [Emscripten port already](https://www.npmjs.com/package/libsodium)?

I tried compiling the node-sodium directly in [Termux app](https://termux.com/) on Android by typing `npm install sodium` but I get the same compile that I get on the Mac. So perhaps there is a package issue too.

**Comment by gardner** [comment]
I have not tried libsodium.js port yet. Performance is a concern but getting it working would definitely be a great step forward. Thanks for that.

It would still be good to know the best practices for getting npm modules with native code cross compiled.

**Comment by lamuertepeluda** [comment]
I see. I'd also like to know how to do it, but remember that [active development for jxcore has sadly been discontinued by Nubisa](http://www.nubisa.com/nubisa-halting-active-development-on-jxcore-platform/). 

If we're lucky, node.js folks will eventually land an officially supported version in the future, [which at the moment lacks](https://github.com/nodejs/node/blob/master/BUILDING.md#android--android-based-devices-eg-firefox-os). There is already [some discussion about it on their repo](https://github.com/nodejs/build/issues/359).

**Comment by dominictarr** [comment]
@lamuertepeluda yes, we are currently using the emscripten port, but it's a lot slower.
We have to verify a lot of signatures, which is the most cpu expensive crypto operation,
so getting libsodium compiled would make a big difference to users.

**Comment by obastemur** [comment]
Which engine you've been using with JXcore? If it was v8 3.14 you may follow leveldown-mobile approach without actually converting V8 API to JXcore macro.

**Comment by gardner** [comment]
I will continue down that path using v8. Thanks @obastemur
